tap_github/client.py

```diff
@@ -64,11 +64,18 @@ def get_next_page_token(
         else:
             results = resp_json.get("items")
 
-        if results:
-            # Paginate as long as the response has items
-            return (previous_token or 1) + 1
+        # Paginate as long as the response has items
+        if not results:
+            return None
+
+        # Test if we have hit the last page for restricted pages such as "events".
```

Comment:
Another option would be to rely on the `next` link directly instead of calculating `next_token` as `previous_token + 1`.

We could do something like this:
```
    next_url = response.links.get('next', {}).get('url')
    parsed = urlparse(url)
    captured_value_list = parse_qs(parsed.query).get('page')
    next_page_string = captured_value_list and captured_value_list[0]
    return int(next_page_string) if next_page_string and next_page_string.isdigit() else None
```

What do you think @aaronsteers ?